quick_start/kusto_sample_app.py

```diff
@@ -0,0 +1,261 @@
+    ingest_connection_string = create_connection_string(ingestUri, authenticationMode)
+    kusto_client = KustoClient(kusto_connection_string)
+    ingest_client = QueuedIngestClient(ingest_connection_string)
+
+    # Todo - Learn More: For additional information on how to create tables see: https://docs.microsoft.com/en-us/azure/data-explorer/one-click-table
+    print("")
+    print(f"Creating table '{databaseName}.{tableName}' if needed:")
+    # Todo (Yochai) what's the table creation command
+    command = ".show version"
+    if not run_control_command(kusto_client, databaseName, command):
+        print("Failed to create or validate table exists.")
+        exit(-1)
+
+    # Todo - Learn More: For additional information on Kusto Query Language see: https://docs.microsoft.com/en-us/azure/data-explorer/write-queries
+    print("")
+    print(f"Initial row count for '{databaseName}.{tableName}' is:")
+    run_query(kusto_client, databaseName, f"{tableName} | summarize count()")
+
+    # Todo - Learn More:
+    #  Kusto batches data for ingestion efficiency. The default batching policy ingests data when one of the following conditions are met:
+    #   1) more then a 1000 files were queued for ingestion for the same table by the same user
+    #   2) more then 1GB of data was queued for ingestion for the same table by the same user
+    #   3) More then 5 minutes have passed since the first file was queued for ingestion for the same table by the same user
+    #  In order to speed up this sample app, we attempt to modify the default ingestion policy to ingest data after 10 seconds have passed
+    #  For additional information on customizing the ingestion batching policy see:
+    #   https://docs.microsoft.com/en-us/azure/data-explorer/kusto/management/batchingpolicy
+    #  You may also skip the batching for some files using the FlushImmediatly property, though this option should be used with care as it is inefficient
+    print("")
+    print(f"Altering the batching policy for '{tableName}'")
+    batching_policy = '{ "MaximumBatchingTimeSpan": "00:00:10", "MaximumNumberOfItems": 500, "MaximumRawDataSizeMB": 1024 }'
+    command = f".alter table {tableName} policy ingestionbatching @'{batching_policy}'"
+    if not run_control_command(kusto_client, databaseName, command):
+        print("Failed to alter the ingestion policy!")
+        print("This could be the result of insufficient permissions.")
+        print("The sample will still run, though ingestion will be delayed for 5 minutes")
+
+    # Todo - Learn More: For additional information on how to ingest data to Kusto in Python see:
+    #  https://docs.microsoft.com/en-us/azure/data-explorer/python-ingest-data
+    print("")
+    ingest_file = input("Please enter a directory of files to ingest from:")
+    print(f"Attempting to ingest '{ingest_file}'")
+    ingest_data_from_file(ingest_client, databaseName, tableName, ingest_file)

```

Comment:
this requires user intervention... should we just ingest a sample? (one that works against the table created about)

Reply:
This is supposed to integrate with one-click so the table schema will not be fixed.
In theory we can ingest the sample provided for oneClickm but that would mean the same data is ingested in each run, so not sure if this is a good idea.